x-pack/test/saved_object_tagging/functional/tests/visualize_integration.ts

```diff
@@ -153,7 +163,10 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
         await selectSavedObjectTags('tag-2');
 
         await testSubjects.click('confirmSaveSavedObjectButton');
+        await PageObjects.common.waitForSaveModalToClose();
+
         await PageObjects.visualize.gotoVisualizationLandingPage();
+        await listingTable.waitUntilTableIsLoaded();
```

Comment:
I still don't understand whether we can get rid of this instruction in the test suit. As suggested in https://github.com/elastic/kibana/pull/82930#discussion_r519762167
IMO, It's okay in the preparatory phase but hard to debug in the test suite, so people start adding them all over the places.

Reply:
There were some rare race conditions where the elements we are manipulating in the next instructions of the test were already removed from the DOM, because we retrieved them before the table was loaded (which cause a rerender, causing some existing dom elements to be removed from the document). This additional check is meant to fix the race by waiting for the next refresh